### PR TITLE
Add Drag & Drop styling for category position column

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add styling for Drag & Drop functionality in Categories list. See screenshots below. 
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | When repositioning categories within the list, it should have new styling. See screenshots below. :warning: Assets may need to be rebuilt.

### New look of positions

screenshot from 2019-02-26 13-17-46

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12570)

### CODE REVIEW

**Comment by sarjon** [comment]
@matks take a look at this PR and let me know what you think.

File: src/Core/Grid/Definition/Factory/CategoryGridDefinitionFactory.php [hunk 1]
```diff
@@ -164,17 +165,20 @@ protected function getColumns()
...
-                (new CategoryPositionColumn('position'))
-                ->setName($this->trans('Position', [], 'Admin.Global'))
-                ->setOptions([
-                    'field' => 'position',
-                    'id_field' => 'id_category',
-                    'id_parent_field' => 'id_parent',
-                    'update_route' => 'AdminCategories',
-                ])
-            );
+            $columns
```
Comment (mickaelandrieu) [inline src/Core/Grid/Definition/Factory/CategoryGridDefinitionFactory.php:164]: it's so smooth API <3

**Review by mickaelandrieu — DISMISSED** [review]

**Comment by marionf** [comment]
@sarjon 

I have a lot of returns because the [previous PR](https://github.com/PrestaShop/PrestaShop/pull/10160) had not been tested :sweat_smile: 


Here the action on the link is ok because the main action is edit

capture d ecran_1045

1) But here the action on the link should be view as the main action of the line 

capture d ecran_1044


2) The edit link in the breadcrumb is the one of the legacy page instead of symfony one

capture d ecran_1057


3) The best category is always changing when I refresh the page

capture d ecran_1047

In the same store, on legacy page the best category is Men

capture d ecran_1048


4) Can we align the ID field and label to the checkbox ?

capture d ecran_1058


5) Can we remove the background color on the hover here:

capture d ecran_1056


6) The materiel icon should be blue on the hover

capture d ecran_1059

capture d ecran_1060


7) The border around the position should be removed

capture d ecran_1061


8) When I drag & drop a category nothing happens and the position isn't changed

https://drive.google.com/file/d/1i2qMmwtTAwLne7bNWj6afA9zStkpdI8R/view?usp=sharing

**Comment by sarjon** [comment]
hey @marionf 

>  The edit link in the breadcrumb is the one of the legacy page instead of symfony one

Categories page is not finished, it will be fixed in other PR

>  The best category is always changing when I refresh the page

i'll check that because it suppose to be using legacy behavior, but either way, it is also part of another PR

>  Can we align the ID field and label to the checkbox ?

i've followed Zeplin prototype, it seems that we need space where "drag" icon can appear.

>  The materiel icon should be blue on the hover

its different than in Zeplin prototype, are you sure?. :/

>  The border around the position should be removed

same here, its different than in Zeplin prototype, are you sure? :/

>  When I drag & drop a category nothing happens and the position isn't changed

i think you have rebuilt assets, maybe hard refresh (CTRL +F5) would help?

**Comment by marionf** [comment]
@sarjon Yes, I checked that with @TristanLDD 

> i think you have rebuilt assets, maybe hard refresh (CTRL +F5) would help?

I tried everything: clear cache, private navigation but it's not saved

**Comment by sarjon** [comment]
> Yes, I checked that with @TristanLDD

its regarding 6 and 7? i should implement them with your comments?

**Comment by marionf** [comment]
> its regarding 6 and 7? i should implement them with your comments?

It's about 4, 6 and 7
Do you need mock-ups ?

**Comment by sarjon** [comment]
> Do you need mock-ups ?

yes, please. :)

**Comment by marionf** [comment]
Let's ask to @TristanLDD :)

File: src/Core/Grid/Column/Type/Common/DraggableColumn.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
```
Comment (matks) [inline src/Core/Grid/Column/Type/Common/DraggableColumn.php:3]: 2019 😉

Comment (sarjon) [reply src/Core/Grid/Column/Type/Common/DraggableColumn.php:3]: :man_facepalming:

**Comment by sarjon** [comment]
@matks do you think we can get prototype?

**Comment by TristanLDD** [comment]
Hello @sarjon, here are the mockups for points 4, 6 and 7: 

4- Align ID:
capture d ecran 2019-02-25 a 15 39 50

6- Color on hover:
capture d ecran 2019-02-25 a 15 40 11

7- Remove border on position:
capture d ecran 2019-02-25 a 15 41 02

Also, I'm noticing that the place holders are missing on the search inputs 

Thank you !

**Comment by sarjon** [comment]
@marionf updated design of positions. Before testing, assets needs to be rebuilt.

**Comment by marionf** [comment]
Thank you @sarjon :+1: 

> 1. But here the action on the link should be view as the main action of the line
53033246-f1401100-3470-11e9-80f1-acd3f9733794

> 5. Can we remove the background color on the hover here:
53085148-8e975580-3502-11e9-902a-aa205ab4692f

Do we fix point 1) and 5) now or in another PR ?

**Comment by sarjon** [comment]
> Do we fix point 1) and 5) now or in another PR ?

@marionf I've fixed both here.

**Comment by marionf** [comment]
@sarjon 

The link should be the same as the action of the line:
If there is a sub category, it's view
If there isn't a sub category, it's edit

capture d ecran_1096

**Comment by sarjon** [comment]
@marionf i was following legacy behavior. Every time you click on row (in new list "Name" column), you are redirected to "View" action. It doesn't matter if category has children or not. Are we suppose to change it? :thinking:

**Comment by marionf** [comment]
Let's ask to @TristanLDD :)

**Comment by TristanLDD** [comment]
@sarjon I thought it was a omission but if it's not let's keep this behavior so it's harmonized with all the other similar pages

**Comment by sarjon** [comment]
hey @marionf seems @TristanLDD agrees to keep current behavior. Is anything else needed for this PR?

**Comment by marionf** [comment]
It's all good @sarjon thanks !

File: src/Core/Grid/Column/Type/Common/DraggableColumn.php (hunk 2 above)
Comment (matks) [reply src/Core/Grid/Column/Type/Common/DraggableColumn.php:3]: It's still here 😝

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thanks @sarjon 😄

**Pull request merged**